src/sentry/incidents/subscription_processor.py

```diff
@@ -416,11 +309,30 @@ def process_update(self, subscription_update: QuerySubscriptionUpdate) -> None:
                 },
             )
 
+        has_metric_alert_processing = features.has(
+            "organizations:workflow-engine-metric-alert-processing", organization
+        )
+        comparison_delta = None
+
+        if has_metric_alert_processing:
+            try:
+                detector = Detector.objects.get(
+                    data_sources__source_id=str(self.subscription.id),
+                    data_sources__type=DATA_SOURCE_SNUBA_QUERY_SUBSCRIPTION,
+                )
+            except Detector.DoesNotExist:
+                logger.exception(
+                    "Detector not found", extra={"subscription_id": self.subscription.id}
+                )
```

Comment:
I wasn't sure if we wanted to do something else here, if we can't find a detector for the query subscription that is pretty bad 🤔 but at this stage I'm just trying to get the comparison delta